File: tiq-e/math/3/main.ts
import {performance} from "perf_hooks";

function isPowerOfThree(n: number): boolean {
    return (Math.log(n) / Math.log(3)) % 1 < 0.00000000000001;
}

const n = 27;

const start = performance.now();
const result = isPowerOfThree(n);
const end = performance.now();
console.log("Duration:", end - start);
console.log("Result:", result);

File: tiq-e/math/3/main.py
import math
from time import perf_counter


class Solution:
    def isPowerOfThree(self, n: int) -> bool:
        if n < 1:
            return False
        return round(math.log(n, 3), 9) == round(math.log(n, 3))


n = 27

start = perf_counter()
result = Solution().isPowerOfThree(n)
end = perf_counter()
print("Duration: " + str(end - start))
print("Result: " + str(result))
